sync/sync.go

```diff
@@ -258,90 +251,114 @@ func (s *Synchronizer) handlePluginRevertBlock() {
+			resetStreams()
+		}
 	}
+
+	s.listener.OnSyncStepDone(OpVerify, committedBlock.Block.Number, time.Since(verifyTimer))
 	return func() {
-		select {
-		case <-ctx.Done():
-			return
-		default:
-			defer close(persisted)
+		s.storeTask(ctx, committedBlock, resetStreams, commitments)
+	}
+}
 
-			if err != nil {
-				s.log.Warnw("Sanity checks failed", "number", block.Number, "hash", block.Hash.ShortString(), "err", err)
-				resetStreams()
-				return
-			}
-			storeTimer := time.Now()
-			err = s.blockchain.Store(block, commitments, stateUpdate, newClasses)
-			if err != nil {
-				if errors.Is(err, blockchain.ErrParentDoesNotMatchHead) {
-					// revert the head and restart the sync process, hoping that the reorg is not deep
-					// if the reorg is deeper, we will end up here again and again until we fully revert reorged
-					// blocks
-					if s.plugin != nil {
-						s.handlePluginRevertBlock()
-					}
-					s.revertHead(block)
+func (s *Synchronizer) storeTask(
+	ctx context.Context,
+	committedBlock *CommittedBlock,
+	resetStreams context.CancelFunc,
+	commitments *core.BlockCommitments,
+) {
+	defer close(committedBlock.Persisted)
+	select {
+	case <-ctx.Done():
+		return
+	default:
+	}
 
-					// The previous head has been reverted, hence, get the current head and store empty pending block
-					head, err := s.blockchain.HeadsHeader()
-					if err != nil {
-						s.log.Errorw("Failed to retrieve the head header", "err", err)
-					}
+	storeTimer := time.Now()
+	block := committedBlock.Block
+	stateUpdate := committedBlock.StateUpdate
+	newClasses := committedBlock.NewClasses
+	if err := s.blockchain.Store(block, commitments, stateUpdate, newClasses); err != nil {
+		if errors.Is(err, blockchain.ErrParentDoesNotMatchHead) {
+			s.revertTask(block, resetStreams)
+			return
+		}
 
-					if head != nil {
-						s.storeEmptyPendingData(head)
-					}
-				} else {
-					s.log.Warnw("Failed storing Block", "number", block.Number,
-						"hash", block.Hash.ShortString(), "err", err)
-				}
-				resetStreams()
-				return
-			}
+		s.log.Warnw("Failed storing Block", "number", block.Number,
+			"hash", block.Hash.ShortString(), "err", err)
+		resetStreams()
+		return
+	}
 
-			s.storeEmptyPendingData(block.Header)
-			s.listener.OnSyncStepDone(OpStore, block.Number, time.Since(storeTimer))
+	s.storeEmptyPendingData(block.Header)
+	s.listener.OnSyncStepDone(OpStore, block.Number, time.Since(storeTimer))
 
-			highestBlockHeader := s.highestBlockHeader.Load()
-			if highestBlockHeader != nil {
-				isBehind := highestBlockHeader.Number > block.Number+uint64(maxWorkers())
-				if s.catchUpMode != isBehind {
-					resetStreams()
-				}
-				s.catchUpMode = isBehind
-			}
+	highestBlockHeader := s.highestBlockHeader.Load()
+	if highestBlockHeader != nil {
+		isBehind := highestBlockHeader.Number > block.Number+uint64(maxWorkers())
+		if s.catchUpMode != isBehind {
+			resetStreams()
+		}
+		s.catchUpMode = isBehind
```

Comment:
very minor nitpick. I understand that it is the same but I believe this suggestion is slightly more readable. 

```suggestion
		if s.catchUpMode != isBehind {
		    s.catchUpMode = isBehind
			resetStreams()
		}
```

Reply:
Updated